packages/store/ts/codegen/field.ts

```diff
@@ -82,27 +82,47 @@ export function renderFieldMethods(options: RenderTableOptions) {
       const dynamicSchemaIndex = schemaIndex - options.staticFields.length;
 
       if (options.withGetters) {
-        result += renderWithFieldSuffix(options.withSuffixlessFieldMethods, field.name, (_methodNameSuffix) =>
-          renderWithStore(
-            storeArgument,
-            ({ _typedStore, _store, _commentSuffix, _methodNamePrefix }) => `
+        if (field.typeWrappingData && field.typeWrappingData.kind === "staticArray") {
+          result += renderWithFieldSuffix(options.withSuffixlessFieldMethods, field.name, (_methodNameSuffix) =>
+            renderWithStore(
+              storeArgument,
+              ({ _typedStore, _commentSuffix, _methodNamePrefix }) => `
               /**
                * @notice Get the length of ${field.name}${_commentSuffix}.
                */
               function ${_methodNamePrefix}length${_methodNameSuffix}(${renderArguments([
-              _typedStore,
-              _typedTableId,
-              _typedKeyArgs,
-            ])}) internal view returns (uint256) {
-                ${_keyTupleDefinition}
-                uint256 _byteLength = ${_store}.getDynamicFieldLength(_tableId, _keyTuple, ${dynamicSchemaIndex});
-                unchecked {
-                  return _byteLength / ${portionData.elementLength};
-                }
+                _typedStore,
+                _typedTableId,
+                _typedKeyArgs,
+              ])}) internal pure returns (uint256) {
+                return ${field.typeWrappingData?.staticLength};
```

Comment:
```suggestion
                return ${field.typeWrappingData.staticLength};
```

Reply:
This gives a type error! I have no idea why the `if` statement isn't "in scope"

![image](https://github.com/latticexyz/mud/assets/29184158/95c0c84a-16b8-4ecd-bc4f-d5c3ac7f1c17)

Reply:
that's annoying but I think it makes sense, since the we're passing in a function and aren't in control of when that function is called

Reply:
could be worth caching it in the conditional and using the cached var, just to make sure we're gonna render what we expected